src/runtime/server/parse-body.ts

```diff
@@ -5,10 +5,17 @@ import contentType, { ParsedMediaType } from 'content-type'
 import { Either, isLeft, left, right } from 'fp-ts/lib/Either'
 import { IncomingMessage } from 'http'
 import httpError, { HttpError } from 'http-errors'
-import querystring from 'querystring'
+import querystring, { ParsedUrlQuery } from 'querystring'
 import getBody from 'raw-body'
+import * as url from 'url'
 import zlib, { Gunzip } from 'zlib'
 
+export function parseQuery(request: IncomingMessage): Either<HttpError, ParsedUrlQuery> {
+  const urlData = (request.url && url.parse(request.url, true).query) || {}
+
+  return right(urlData)
+}
```

Comment:
Yeah but returning a single `Right` was causing problems. And returning just the value was also causing friction because `parseBody` is returning an `Either` as well, and TS couldn't figure it out anymore to narrow down the union when the type was `Either<Error, X> | Y`